docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx

```diff
@@ -0,0 +1,335 @@
+            </a>
+          </li>
+          <li>You're not using Dagster's GitHub app</li>
+          <li>
+            You want to manage environment variables outside of Dagster Cloud
+          </li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <a href="/dagster-cloud/developing-testing/environment-variables/setup-for-hybrid-deployments">
+          Agent configuration
+        </a>
+      </td>
+      <td>
+        Environment variables are defined in the agent's configuration in your
+        project's <code>dagster_cloud.yaml</code> file.{" "}
+        <strong>Supported for Hybrid deployments only</strong>.
+      </td>
+      <td>
+        <ul
+          style={{
+            marginTop: "0px",
+          }}
+        >
+          <li
+            style={{
+              marginTop: "0px",
+            }}
+          >
+            You're using{" "}
+            <a href="/dagster-cloud/deployment/hybrid">Hybrid deployment</a>
+          </li>
+          <li>
+            You want to manage environment variables outside of Dagster Cloud
+          </li>
+        </ul>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Storage and encryption
+
+This may only apply to management using the UI. Will need to confirm once details are finalized.
+
+<TODO></TODO>
+
+### Reserved variables
+
+[Built-in Dagster Cloud environment variables](#built-in-environment-variables) are reserved and are unavailable for use. If used, the following error will be raised: <TODO></TODO>
+
+---
+
+## Managing environment variables
+
+The simplest way to manage environment variables is to use Dagster Cloud's built-in secrets manager, which allows you to create and manage environment variables right in the UI. If you prefer not to use Dagster Cloud's secrets manager, you can use one of [the other methods](#using-alternative-methods) supported for your deployment type.
+
+### Using the Dagster Cloud UI
+
+<Note>
+  <a href="/dagster-cloud/account/managing-users">
+    <strong>Organization</strong>, <strong>Admin</strong>, or{" "}
+    <strong>Editor</strong> permissions
+  </a>{" "}
+  are required to manage environment variables in Dagster Cloud. This includes
+  viewing the value of an existsing environment variable.
+</Note>
+
+Declaring environment variables in the Dagster Cloud UI is supported for both Serverless and Hybrid deployments.
+
+1. Sign into your Dagster Cloud account.
+2. Click the code location you want to add an environment variable to.
+3. Click the **Environment Variables** tab.
+4. Click **+ Add** to add a new environment variable.
+5. In the window that displays, fill in the following:
+   - **Key name** - Enter a name for the environment variable. This is how the variable will be referenced in your code.
+   - **Value** - Enter a value for the environment variable.
+   - Select the deployment(s) to apply the variable to:
+     - **Production**
+     - **Branch Deployments**
+     - **Local**
```

Comment:
you will also be able to optionally choose to apply it to specific code locations or to all of them